### ISSUE TITLE
Attachments to transactions are not persisted if sent within the same envelope

### ISSUE THREAD

**Issue by jernejstrasner** [getsentry/relay#1761]
If an envelope with a transaction event and attachments is sent, the attachments are lost. If the attachment is sent via the dedicated attachments endpoint, everything is fine.

The reason for this is that we don't do any attachments processing in save_transaction_event. The attachments are written in processing store, and only in save_event (error events case) we actually copy them to persistent storage.

The fix is (in my opinion) NOT to introduce attachments processing in the transaction processing pipeline the same way it happens for errors. That is something we do because we need to mutate the event based on data found in attachments (eg stacktrace processing), but it causes lots of small operational problems such as high variance in timing of save_event.

Instead, Relay should treat attachments to transaction events as individual attachments, and write them to kafka that way. Ideally it would do so for all attachments that are not processing-relevant, even for error events.

[transaction-with-attachment.json.txt](https://github.com/getsentry/relay/files/10458746/transaction-with-attachment.json.txt)

**Comment by jjbayer** [comment]
- [x] Do we drop transactions anywhere in sentry ingestion (e.g. save_event)? In that case, attachments should also be dropped. @jjbayer.

Answer: we log an Outcome.FILTERED in `discard_event` and `filter_attachments_for_group`, but both of these are called _after_ the early return for transactions:

https://github.com/getsentry/sentry/blob/2c126ce9d6b41df32c8ccaeed84ccc45e7dd140d/src/sentry/event_manager.py#L442
 
TL;DR transactions are not being filtered on sentry side.
